### PR TITLE
create network policy on challenge create

### PR DESCRIPTION
**Pull request by sroettger**

### CODE REVIEW

File: kctf-operator/go.sum [hunk 1]
```diff
@@ -497,6 +497,7 @@ github.com/hashicorp/memberlist v0.1.4/go.mod h1:ajVTdAv/9Im8oMAAj5G31PhhMCZJV2p
 github.com/hashicorp/memberlist v0.1.5/go.mod h1:ajVTdAv/9Im8oMAAj5G31PhhMCZJV2pPBoIllUwCN7I=
 github.com/hashicorp/serf v0.8.2/go.mod h1:6hOLApaqBFA1NXqRQAsxw9QxuDEvNxSQRwA/JwenrHc=
 github.com/hashicorp/serf v0.8.5/go.mod h1:UpNcs7fFbpKIyZaUuSW6EPiH+eZC7OuyFD+wc1oal+k=
+github.com/hpcloud/tail v1.0.0 h1:nfCOvKYfkgYP8hkirhJocXT2+zOD8yUNjXaWfTlyFKI=
```
Comment (sroettger) [inline kctf-operator/go.sum:500]: I don't know what these are. These got auto updated :)

File: kctf-operator/pkg/resources/initializer.go [hunk 2]
```diff
@@ -17,15 +17,13 @@ var log logr.Logger = logf.Log.WithName("cmd")
 func InitializeOperator(client *client.Client) error {
 	// Creates the objects that enable the DNS, external DNS and etc
 
-	objectFunctions := []func() runtime.Object{NewApparmorProfiles, NewAllowDns, NewServiceAccountGcsFuseSa,
+	objectFunctions := []func() runtime.Object{NewApparmorProfiles, NewServiceAccountGcsFuseSa,
```
Comment (sroettger) [inline kctf-operator/pkg/resources/initializer.go:20]: get rid of the network policy creation in the initializer

File: kctf-operator/pkg/controller/challenge/challenge_controller.go [hunk 3]
```diff
@@ -177,7 +178,7 @@ func updateConfigurations(challenge *kctfv1alpha1.Challenge, cl client.Client, s
 	log logr.Logger, ctx context.Context) (bool, error) {
 	// We check if there's an error in each update
 	updateFunctions := []func(challenge *kctfv1alpha1.Challenge, client client.Client, scheme *runtime.Scheme,
-		log logr.Logger, ctx context.Context) (bool, error){volumes.Update,
+		log logr.Logger, ctx context.Context) (bool, error){network.Update, volumes.Update,
```
Comment (sroettger) [inline kctf-operator/pkg/controller/challenge/challenge_controller.go:181]: add the network policy update code when creating/updating a challenge

File: kctf-operator/go.sum (hunk 1 above)
Comment (sirdarckcat) [reply kctf-operator/go.sum:500]: Backdoors

**Review by sirdarckcat — APPROVED** [review]
How much of this is boilerplate?

**Pull request merged**